internal/controller/common.go

```diff
@@ -75,3 +78,50 @@ func getGitProvider(component appstudiov1alpha1.Component) (string, error) {
 
 	return gitProvider, err
 }
+
+// Get only components that match a given workspace/application/componentname
+func getFilteredComponents(workspaces []mmv1alpha1.WorkspaceSpec, apiClient client.Client, ctx context.Context) ([]appstudiov1alpha1.Component, error) {
+	components := []appstudiov1alpha1.Component{}
+	err := error(nil)
+
+	// Iterate workspaces and create query filtered by namespace ({workspace}-tenant)
+	for _, workspace := range workspaces {
+		workspaceComponentList := &appstudiov1alpha1.ComponentList{}
+		listOps := &client.ListOptions{
+			Namespace: workspace.Workspace + "-tenant",
```

Comment:
I'm unsure if this is something hard-coded in Konflux or it's configurable, if this is configurable, this might not work for other konflux clusters (for example the one setup for Fedora community). I think it's fine at this moment, please take a note to recheck this once we have some important features relies on such kind of CRs (for example when we support component nudging in mintmaker).